src/config/server.rs

```diff
@@ -47,3 +48,13 @@ impl Server {
 pub struct Proxy {
   pub url: String,
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct HttpSettings {
+  pub keep_alive_interval: u64,
+  pub keep_alive_timeout: u64,
+  pub keep_alive_while_idle: bool,
+  pub pool_idle_timeout: u64,
+  pub pool_max_idle_per_host: usize,
```

Comment:
I think we should try to support all the settings provided over here —  https://github.com/seanmonstar/reqwest/blob/master/src/async_impl/client.rs#L87


```rust
struct Config {
    // NOTE: When adding a new field, update `fmt::Debug for ClientBuilder`
    accepts: Accepts,
    headers: HeaderMap,
    #[cfg(feature = "native-tls")]
    hostname_verification: bool,
    #[cfg(feature = "__tls")]
    certs_verification: bool,
    #[cfg(feature = "__tls")]
    tls_sni: bool,
    connect_timeout: Option<Duration>,
    connection_verbose: bool,
    pool_idle_timeout: Option<Duration>,
    pool_max_idle_per_host: usize,
    tcp_keepalive: Option<Duration>,
    #[cfg(any(feature = "native-tls", feature = "__rustls"))]
    identity: Option<Identity>,
    proxies: Vec<Proxy>,
    auto_sys_proxy: bool,
    redirect_policy: redirect::Policy,
    referer: bool,
    timeout: Option<Duration>,
    #[cfg(feature = "__tls")]
    root_certs: Vec<Certificate>,
    #[cfg(feature = "__tls")]
    tls_built_in_root_certs: bool,
    #[cfg(feature = "__tls")]
    min_tls_version: Option<tls::Version>,
    #[cfg(feature = "__tls")]
    max_tls_version: Option<tls::Version>,
    #[cfg(feature = "__tls")]
    tls_info: bool,
    #[cfg(feature = "__tls")]
    tls: TlsBackend,
    http_version_pref: HttpVersionPref,
    http09_responses: bool,
    http1_title_case_headers: bool,
    http1_allow_obsolete_multiline_headers_in_responses: bool,
    http1_ignore_invalid_headers_in_responses: bool,
    http1_allow_spaces_after_header_name_in_responses: bool,
    http2_initial_stream_window_size: Option<u32>,
    http2_initial_connection_window_size: Option<u32>,
    http2_adaptive_window: bool,
    http2_max_frame_size: Option<u32>,
    http2_keep_alive_interval: Option<Duration>,
    http2_keep_alive_timeout: Option<Duration>,
    http2_keep_alive_while_idle: bool,
    local_address: Option<IpAddr>,
    nodelay: bool,
    #[cfg(feature = "cookies")]
    cookie_store: Option<Arc<dyn cookie::CookieStore>>,
    trust_dns: bool,
    error: Option<crate::Error>,
    https_only: bool,
    #[cfg(feature = "http3")]
    tls_enable_early_data: bool,
    #[cfg(feature = "http3")]
    quic_max_idle_timeout: Option<Duration>,
    #[cfg(feature = "http3")]
    quic_stream_receive_window: Option<VarInt>,
    #[cfg(feature = "http3")]
    quic_receive_window: Option<VarInt>,
    #[cfg(feature = "http3")]
    quic_send_window: Option<u64>,
    dns_overrides: HashMap<String, Vec<SocketAddr>>,
    dns_resolver: Option<Arc<dyn Resolve>>,
}
```

Reply:
@tusharmath  Are all the fields optional? And as we are using a builder pattern for initializing the client options.  So I have to make a check for each and every field and instaniate? Or is there any other way? 

something like this 
```rust
if let Some(ref upstream) = server.upstream {
      if let Some(ref keep_alive_interval) = upstream.keep_alive_interval {
        builder = builder.http2_keep_alive_interval(Duration::from_millis(keep_alive_interval.clone()))
      }
     // similarly for other fields
 }

```

Reply:
You could do something as follows — 

```rust
// config::Server.rs
Upstream {
  keep_alive_timeout: u64
  // and the rest of the fields ...
}

impl From<Upstream> for reqwest::Config {
  fn from(value: Upstream) -> Self {
    reqwest::Config {
      // ... now set fields one by one
    }
  }
}
```

Once you have the reqwest::Config, you can use that directly to create a client.

Reply:
```
 cannot find type 'Config' in crate 'reqwest' not found in 'reqwest'rustc
```

Getting this compilation error.

Reply:
I think only way to set custom configuration is through builder() ( Maybe I m wrong , I m new to rust )

https://rcos.io/static/internal_docs/reqwest/struct.Client.html
> The Client has various configuration values to tweak, but the defaults are set to what is usually the most commonly desired value. To configure a Client, use Client::builder().

Reply:
Just realized, that Config is private inside of reqwest, we will have to take another approach.